files/rsyslog/rsyslog.sh

```diff
@@ -6,26 +6,115 @@ export LOGGING_FILE_PATH=${LOGGING_FILE_PATH:-"/var/log/rsyslog/rsyslog.log"}
 export RSYSLOG_WORKDIRECTORY=${RSYSLOG_WORKDIRECTORY:-/var/lib/rsyslog.pod}
 export MERGE_JSON_LOG=${MERGE_JSON_LOG:-false}
 export UNDEFINED_DEBUG=${UNDEFINED_DEBUG:-false}
-# if false, mmexternal undefined_field won't be called.
-export USE_MMEXTERNAL=${USE_MMEXTERNAL:-true}
-# if true, mmnormalize skipempty is executed.
-# effective when USE_MMEXTERNAL is false.
-export SKIP_EMPTY=${SKIP_EMPTY:-false}
 export RSYSLOG_FILE_READ_FROM_TAIL=${RSYSLOG_FILE_READ_FROM_TAIL:-off}
 export RSYSLOG_JOURNAL_READ_FROM_TAIL=${RSYSLOG_JOURNAL_READ_FROM_TAIL:-off}
 
-# Create a config file for undefined fields from env vars
+# Undefined field configuraion
+CDM_USE_UNDEFINED=${CDM_USE_UNDEFINED:-false}
+CDM_KEEP_EMPTY_FIELDS=${CDM_KEEP_EMPTY_FIELDS:-""}
+CDM_UNDEFINED_TO_STRING=${CDM_UNDEFINED_TO_STRING:-false}
+CDM_UNDEFINED_DOT_REPLACE_CHAR=${CDM_UNDEFINED_DOT_REPLACE_CHAR:-"UNUSED"}
+if [ "${CDM_USE_UNDEFINED}" = "false" -a "${CDM_KEEP_EMPTY_FIELDS}" = "" -a "${CDM_UNDEFINED_TO_STRING}" = "false" -a "${CDM_UNDEFINED_DOT_REPLACE_CHAR}" = "UNUSED" ] ; then
+    if [ "${SKIP_EMPTY:-}" = "" ]; then
+        if [ "${USE_MMEXTERNAL:-}" = "" ]; then
+            # default - mmnormalize skip-empty
+            export SKIP_EMPTY=true
+            export USE_MMEXTERNAL=false
+        elif [ "${USE_MMEXTERNAL,,}" = "true" ]; then
+            export SKIP_EMPTY=false
+            export USE_MMEXTERNAL=true
+            echo "INFO: USE_MMEXTERNAL was set but none of the UNDEFINED fields were set. Consider using SKIP_EMPTY=true instead."
+        else
+            export SKIP_EMPTY=true
+            export USE_MMEXTERNAL=false
+            echo "INFO: SKIP_EMPTY was set but USE_MMEXTERNAL was set to false. Enabling SKIP_EMPTY."
```

Comment:
`INFO: SKIP_EMPTY was not set but USE_MMEXTERNAL was set to false. Enabling SKIP_EMPTY.  Consider unsetting USE_MMEXTERNAL to avoid this message in the future.`

Reply:
I guess I'd better append "to avoid this message in the future" to each INFO message?

Reply:
Yes, good idea, in case the user is wondering how to get rid of this message at startup . . .